stonesoup/models/measurement/nonlinear.py

```diff
@@ -965,3 +967,192 @@ def rvs(self, num_samples=1, **kwargs) -> Union[StateVector, StateVectors]:
         out = super().rvs(num_samples, **kwargs)
         out = np.array([[Elevation(0)], [Bearing(0)], [0.], [0.]]) + out
         return out
+
+
+class RangeRangeRateBinning(CartesianToElevationBearingRangeRate):
+    r"""This is a class implementation of a time-invariant measurement model, \
+    where measurements are assumed to be in the form of elevation \
+    (:math:`\theta`),  bearing (:math:`\phi`), range (:math:`r`) and
+    range-rate (:math:'\dot{r}'), with Gaussian noise in each dimension and the
+    range and range-rate are binned based on the
+    range resolution and range-rate resolution respectively.
+
+    The model is described by the following equations:
+
+    .. math::
+
+      \vec{y}_t = h(\vec{x}_t, \vec{v}_t)
+
+    where:
+
+    * :math:`\vec{y}_t` is a measurement vector of the form:
+
+    .. math::
+
+      \vec{y}_t = \begin{bmatrix}
+                \theta \\
+                \phi \\
+                r \\
+                \dot{r}
+            \end{bmatrix}
+
+    * :math:`h` is a non-linear model function of the form:
+
+    .. math::
+
+      h(\vec{x}_t,\vec{v}_t) = \begin{bmatrix}
+                asin(\mathcal{z}/\sqrt{\mathcal{x}^2 + \mathcal{y}^2 +\mathcal{z}^2}) \\
+                atan2(\mathcal{y},\mathcal{x}) \\
+                \sqrt{\mathcal{x}^2 + \mathcal{y}^2 + \mathcal{z}^2} \\
+                (x\dot{x} + y\dot{y} + z\dot{z})/\sqrt{x^2 + y^2 + z^2}
+                \end{bmatrix} + \vec{v}_t
+
+    * :math:`\vec{v}_t` is Gaussian distributed with covariance :math:`R`, i.e.:
+
+    .. math::
+
+      \vec{v}_t \sim \mathcal{N}(0,R)
+
+    .. math::
+
+      R = \begin{bmatrix}
+            \sigma_{\theta}^2 & 0 & 0 & 0\\
+            0 & \sigma_{\phi}^2 & 0 & 0\\
+            0 & 0 & \sigma_{r}^2 & 0\\
+            0 & 0 & 0 & \sigma_{\dot{r}}^2
+            \end{bmatrix}
+
+    The range and range rate are binned using
+
+    .. math::
+
+        x = floor(x/\Delta x)*\Delta x
```

Comment:
The binning output gives the value at the starting edge of the bin; e.g. x=0, del_x = 10, then x_bin =0. Are there times when it makes sense to return the value in the middle of the bin?  Additional thought - The range gating of the radar may not align with starting from zero. If the radar starts binning the range at 2025.0 meters with delta range of 10, then the binning won't line up actual radar. @sdhiscocks Perhaps we want some binning functions that can be used here as well as in the radar sim classes?

Reply:
An acceleration model can be used. The mapping and velocity mapping gets the position and velocity from the state so the state can have more than 6 dimensions.

Reply:
So I think the documentation just needs to be modified - right now it explicitly states a 6D state model. I think this part of the documentation is being grabbed from the parent class.

Reply:
Should the documentation instead say it accepts a state with a minimum 6D? Should the other measurement model also be changed too because most measurement model have a minimum number of dimensions needed in a state but no maximum?